modules/RTC/TrackStreamingStatus.js

```diff
@@ -0,0 +1,758 @@
+            // video muted there is no way for the connection to be inactive,
+            // because the detection logic below only makes sense for video.
+            return TrackStreamingStatus.ACTIVE;
+        }
+
+        // Logic when isVideoTrackFrozen is supported
+        if (browser.supportsVideoMuteOnConnInterrupted()) {
+            if (!isVideoTrackFrozen) {
+                // If the video is playing we're good
+                return TrackStreamingStatus.ACTIVE;
+            } else if (isInForwardedSources) {
+                return isRestoringTimedout
+                    ? TrackStreamingStatus.INTERRUPTED
+                    : TrackStreamingStatus.RESTORING;
+            }
+
+            return TrackStreamingStatus.INACTIVE;
+        }
+
+        // Because this browser is incapable of detecting frozen video we must
+        // rely on the forwarded sources value
+        return isInForwardedSources
+            ? TrackStreamingStatus.ACTIVE
+            : TrackStreamingStatus.INACTIVE;
+    }
+
+    /* eslint-enable max-params*/
+
+    /**
+     * In P2P mode we don't care about any values coming from the JVB and
+     * the connection status can be only active or interrupted.
+     * @param {boolean} isVideoMuted true if video muted
+     * @param {boolean} isVideoTrackFrozen true if the video track for
+     * the remote track is currently frozen. If the current browser does not
+     * support video frozen detection then it's always false.
+     * @return {TrackStreamingStatus}
+     * @private
+     */
+    static _getNewStateForP2PMode(isVideoMuted, isVideoTrackFrozen) {
+        if (!browser.supportsVideoMuteOnConnInterrupted()) {
+            // There's no way to detect problems in P2P when there's no video
+            // track frozen detection...
+            return TrackStreamingStatus.ACTIVE;
+        }
+
+        return isVideoMuted || !isVideoTrackFrozen
+            ? TrackStreamingStatus.ACTIVE
+            : TrackStreamingStatus.INTERRUPTED;
+    }
+
+    /**
+     * Creates new instance of <tt>TrackStreamingStatus</tt>.
+     *
+     * @constructor
+     * @param {RTC} rtc the RTC service instance
+     * @param {JitsiConference} conference parent conference instance
+     * @param {Object} options
+     * @param {number} [options.p2pRtcMuteTimeout=2500] custom value for
+     * {@link TrackStreamingStatus.p2pRtcMuteTimeout}.
+     * @param {number} [options.rtcMuteTimeout=2000] custom value for
+     * {@link TrackStreamingStatus.rtcMuteTimeout}.
+     * @param {number} [options.outOfLastNTimeout=500] custom value for
+     * {@link TrackStreamingStatus.outOfLastNTimeout}.
+     */
+    constructor(rtc, conference, track, options) {
+        this.rtc = rtc;
+        this.conference = conference;
+        this.track = track;
+
+        /**
+         * This holds the timeout callback ID scheduled using
+         * window.setTimeout.
+         * @type {number}
+         */
+        this.trackTimer = null;
```

Comment:
`this.trackTimers`, `this.rtcMutedTimestamp`, `this.enteredLastNTimestamp`, `this.restoringTimers`, and `this.connectionStatusMap` are now flattened to no longer hold a mapping of endpointId to value.